moodle/moodle.go

```diff
@@ -410,7 +412,18 @@ func (c *Controller) createDeployment(foo *operatorv1.Moodle) (error, string, st
 	mysqlUserName := foo.Spec.MySQLUserName
 	fmt.Printf("MySQL Username:%v\n", mysqlUserName)
 
-	mysqlUserPassword := foo.Spec.MySQLUserPassword
+	secretsClient := c.kubeclientset.CoreV1().Secrets(namespace)
+	secret, err := secretsClient.Get(foo.Spec.MysqlPasswordSecretName, metav1.GetOptions{})
+
+	if err != nil {
+		fmt.Printf("Error, secret %s, not found from in namespace %s: %v\n", foo.Spec.MysqlPasswordSecretName, namespace, err)
+	}
+	if _, ok := secret.Data["mySQLUserPassword"]; !ok {
```

Comment:
Where is "mySQLUserPassword" field coming from? 

We should not be hardcoding any such fields..

Reply:
`mySQLUserPassword` is the key, that contains the base64 password. 

I have to use some sort of structure or key I thought.. How else could I store the password if not in some arbitrary field? For example, the secret that MySQLCluster reads from, contains the ROOT_PASSWORD field

Reply:
That is true. However, we want to have a generic solution. So one option is in Moodle resource we can take value as secretName.fieldName. Then the Operator code will not have to hard code the fieldName like above. So basically: cluster1-secret.ROOT_PASSWORD instead of just cluster1-secret.

Also, in above case, wouldn't the field be 'ROOT_PASSWORD'? Since that is the field used when creating cluster1-secret.

Reply:
So: 
in moodle1.yaml
`  mysqlPassword: cluster1-secret.ROOT_PASSWORD`
and then inside the code, I split on the "." , query kubeclient to get the secret, look for that field, and decode it right?

And yes you are correct, but I was thinking that we are making two secrets, one for moodle, one for mysqlcluster. I tried to clarify this a little. But then, one more question, if I do make the above change, then how will MysqlCluster get its ROOT_VALUE? Doesn't it expect it to have a field called ROOT_VALUE? Do you understand my question or should I try and reclarify

Reply:
> So something like:
> in moodle1.yaml
> ` mysqlPassword: cluster1-secret.ROOT_PASSWORD`
> and then inside the code, I split on the "." , query kubeclient to get the secret, look for that field, and decode it right?
> 

Yes, this is correct.

> And yes correct, but I was thinking that we are making two secrets, one for moodle, one for mysqlcluster. 

The secret Moodle is created internally by the Moodle controller. On the other hand, the secret for MysqlCluster is something created directly using kubectl. We want to refer to this secret in the Moodle Custom Resource Spec definition. 

But then, one more question, if I do make the above change, then how will MysqlCluster get its ROOT_VALUE? Doesn't it expect it to have a field called ROOT_VALUE? Do you understand my question or should I try and reclarify

I did not follow above point about ROOT_VALUE. Isn't the Secret's created using ROOT_PASSWORD as the key? 
https://github.com/cloud-ark/kubeplus/blob/master/examples/moodle-with-presslabs/namespace1/cluster1-secret.yaml#L9

Reply:
Right, actually thinking about it, the above change to secret-name.ROOT_VALUE should work.